chore(gen2-storage docs): Add documentation for gen2 StorageImage (#5188)
aws-amplify/amplify-ui: docs/src/pages/[platform]/connected-components/storage/react.mdx

import { InstallScripts } from '@/components/InstallScripts';
import { NextSteps } from './NextSteps';

Amplify UI Storage components use [Amplify Storage](https://docs.amplify.aws/react/build-a-backend/storage/) to allow your users to upload and interact with files stored in Amazon S3 with minimal boilerplate.

## Quick start

Follow the steps in [this guide](https://docs.amplify.aws/react/build-a-backend/storage/) to set up your Amplify Storage backend.

<Alert role="none" variation="info" heading="Amplify CLI">

>   Using the Amplify CLI? Check out this [documentation](https://docs.amplify.aws/gen1/react/build-a-backend/storage/).

Reviewer: Please split this into a separate PR.

<InstallScripts component="storage" />

> ## Using Storage with the Authenticator

Reviewer: ```suggestion
## Configuring Amplify for Storage UI Components
```

> To use Storage components with the Authenticator, such as when loading private or protected images, you'll first need to configure Amplify as shown below. 

Reviewer: ```suggestion
To use Storage UI components with Amplify, you'll need to call `Amplify.configure()` as shown below. 
```
